src/Jobs/PtpJob.php

```diff
@@ -0,0 +1,236 @@
+        $this->points = [];
+        $this->labelId= $labelId;
+        $this->annotationIds = [];
+        $this->jobType = $jobType;
+        foreach ($annotations as $annotation) {
+            $this->points[] = implode(',', $annotation['points']);
+            //TODO: find a better way to pass this arg.
+            $this->annotationIds[] = $annotation['annotation_id'];
+        }
+
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $image = Image::findOrFail($this->imageId);
+        $volumeId = $image->volume->id;
+        FileCache::getOnce($image, function ($image, $path) use ($volumeId){
+            $this->python($path, $volumeId);
+        });
+    }
+    /**
+     * Find median of array
+     *
+     * @param  $array
+     * @return integer
+     */
+    protected function findMedian(array $array): int
+    {
+        sort($array);
+        $count = count($array);
+        $index = intdiv($count, 2);
+        if ($count % 2 == 0) {    // count is odd
+            return $array[$index];
+        } else {                   // count is even
+            return ($array[$index-1] + $array[$index]) / 2;
+        }
+    }
+    /**
+     * Run the python script for Point to Polygon conversion
+     *
+     * @param  $imagePath The path the image is found
+     * @param  $volumeId The ID of the volume
+     * @param  $log File where the logs from the python script will be found
+     */
+    protected function python(string $imagePath, int $volumeId, string $log = 'log.txt')
+    {
+        $code = 0;
+        $lines = [];
+        $python = config('ptp.python');
+        $script = config('ptp.ptp_script');
+        $logFile = config('ptp.temp_dir').'/'.$log;
+        $points = implode(' ',$this->points);
+        $labelId = $this->labelId;
+        $device = config('ptp.device');
+        $modelPath = config('ptp.model_path');
+        $modelType = config('ptp.model_type');
+        $imageId = $this->imageId;
+        $annotationIds = implode(' ', $this->annotationIds);
+        $jobType = $this->jobType;
+        $outputFile = $this->outputFile;
+        $command = "{$python} -u {$script} {$jobType} -i {$imagePath} --image-id {$imageId} -p {$points} -l {$labelId}  -a {$annotationIds} --device {$device} --model-type {$modelType} --model-path {$modelPath} --output-file {$outputFile} ";
+        if ($jobType == 'ptp') {
+            $expectedAreaValues = json_decode(PtpExpectedArea::where('label_id', $labelId)->where('volume_id', $volumeId)->first()->areas);
+            $medianArea = $this->findMedian($expectedAreaValues);
+            $command = $command." -e $medianArea";
+        }
+        exec("$command > {$logFile} 2>&1", $lines, $code);
+
+
+        if ($code !== 0) {
+            $lines = File::get($logFile);
+            throw new Exception("Error while executing python script'{$script}':\n{$lines}", $code);
+        }
```

Comment:
There should be no need to specify an extra log file ([example](https://github.com/biigle/magic-sam/blob/925439991751d088ab0b46a543d49772442884ac/src/Jobs/GenerateEmbedding.php#L133-L138)).